hooks/invocation_sample/main.go

```diff
@@ -0,0 +1,203 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+)
+
+// HookCatalog created during module loading, consists of hook implementations
+type HookCatalog struct {
+	entrypointHooks map[string]map[string]EntrypointHook
+	// other hooks
+}
+
+func (c HookCatalog) getEntrypointHook(moduleCode, hookCode string) EntrypointHook {
+	if mh, ok := c.entrypointHooks[moduleCode]; ok {
+		if v, ok := mh[hookCode]; ok {
+			return v
+		}
+	}
+
+	return nil
+}
+
+type EntrypointHook interface {
+	HandleEntrypointHook(req *http.Request, body []byte)
+}
+
+type EntrypointHookImpl struct {
+	callback string // for simplification
+}
+
+func (i EntrypointHookImpl) HandleEntrypointHook(_ *http.Request, _ []byte) {
+	fmt.Println(i.callback)
+}
+
+type HookStageExecutor struct {
+	hostExecutionPlan           ExecutionPlan
+	defaultAccountExecutionPlan ExecutionPlan
+	hookCatalog                 HookCatalog
+}
+
+// ExecutionPlan execution plan hierarchy
+type ExecutionPlan struct {
+	endpoints map[string]EndpointExecutionPlan
+}
+
+type EndpointExecutionPlan struct {
+	stages map[string]StageExecutionPlan
+}
+
+type StageExecutionPlan struct {
+	groups []ExecutionGroup
+}
+
+type ExecutionGroup struct {
+	timeout      int
+	hookSequence []HookId
+}
+
+type HookId struct {
+	moduleCode   string
+	hookImplCode string
+}
+
+// CreateExecutor called during application configuration
+func CreateExecutor(hostExecutionPlan string, defaultAccountExecutionPlan string, hookCatalog HookCatalog) HookStageExecutor {
+	return HookStageExecutor{
+		hostExecutionPlan:           parseAndValidateExecutionPlan(hostExecutionPlan),
+		defaultAccountExecutionPlan: parseAndValidateExecutionPlan(defaultAccountExecutionPlan),
+		hookCatalog:                 hookCatalog,
+	}
+}
+
+type Endpoint string
+
+const (
+	openrtb2Auction Endpoint = "/openrtb2/auction"
+	//other endpoints
+)
+
+func (e Endpoint) String() string {
+	return string(e)
+}
+
+type Stage string
+
+const (
+	entrypoint Stage = "entrypoint"
+	//other stages
+)
+
+func (s Stage) String() string {
+	return string(s)
+}
+
+type Context struct {
+	endpoint Endpoint
+}
```

Comment:
We may need to use a different name instead of Context, or can this double as the Go cancellation context?

Reply:
I think we will use a different name